Run every test with and without scoped-element-registry polyfill

aghArdeshir opened the pull request:
This PR fixes https://github.com/ing-bank/lion/issues/2351

- Runs every test two times: once with the scoped-element-registry polyfill and once without it.
- Uses `testGroups` to achieve this
- Why? Because that polyfill affects the behavior of components, and gives us false positives. The users of the lion library may not all have loaded that polyfill.
- Creates two suffixes for test files: `some-test.polyfill.test.js` and `some-test.no-polyfill.test.js` for tests that require to be run only with/without the polyfill. (Example is the SlotMixin tests)
- The SlotMixin test had two tests at the end of the file:
- - Moved one of them to another file and suffixed it with `.polyfill`, because that needed polyfill to pass. Also refactored it to use a simple `spy` instead of a mocked object.
- - Removed the other test because it was lying. The title of the test was `does not scope elements when polyfill not loaded` but when you actually remove the polyfill from test config, it fails! (I gave more explanation here: https://github.com/ing-bank/lion/issues/2358)
- - :question: Do I need to extract the refactor and removal of SlotMixin tests as a separate MR?
- :question:  I'd loved to be able to hint when a test fails, did it fail with or without the polyfill, but failed to do so. Any idea?

**UPDATE: Below is the original message of MR, which is not accurate anymore. This PR is re-purposed.**

~~- We should not load a polyfill globally in tests environment, because it affects how tests behave and may give us false positive and false confidence.~~
~~- Every test that needs polyfill, loads the polyfill only for itself.~~
~~- No unnecessary mocking is required. Spy would do the job in this scneario.~~
~~- Adds a util for adding a script in tests on demand.~~
~~- Removes one unnecessary ts-ignore~~

okadurin commented:
Hi @aghArdeshir ,
Thanks a lot for the contribution! Indeed at some point we added a polyfill for all tests. The right solution would be to run all the tests with and without the polyfill to be sure the code works well in all cases. I think ideally there should be 2 configuration files like `web-test-runner-with-polyfill.config.mjs` and `web-test-runner-without-polyfill.config.mjs` which we run on every `npm run test`. To do that it will require duplicating some tests as some of them were slightly modified when the polyfill support was added. [Here](https://github.com/ing-bank/lion/commit/2fab9e5559a570b7e47f773846cb8cf45e6184f7) is one of those that was modified. 
Do you think you could propose a solution for this request?

aghArdeshir commented:
Intersting @okadurin . I see now what we are trying to test. Yes I may be able to look into this.

I'll let you know here if I could come up with a solution or if I eventually gave up

aghArdeshir commented:
@okadurin I've updated my MR and have also updated my original message (https://github.com/ing-bank/lion/pull/2353#issue-2490418176) to list what changes I made. Take another look?

tlouisse commented:
@aghArdeshir would you like to create a PR for this?

tlouisse commented:
@aghArdeshir I added a solution for this and all related tickets here: https://github.com/ing-bank/lion/pull/2407/commits
Thanks again for the help! 👍

tlouisse commented:
https://github.com/ing-bank/lion/pull/2407

aghArdeshir commented:
Hi. Saw your message late, but I see all is taken care of now :smiley: Thanks
